src/config.js

```diff
@@ -0,0 +1,47 @@
+/* @flow */
+import requireUncached from 'require-uncached';
+
+import {createLogger} from './util/logger';
+import {UsageError} from './errors';
+
+const log = createLogger(__filename);
+
+type ApplyConfigToArgvParams = {|
+  argv: Object,
+  configObject: Object,
+  defaultValues: Object,
+|};
+
+export function applyConfigToArgv({
+  argv,
+  configObject,
+  defaultValues = {},
+}: ApplyConfigToArgvParams): Object {
+  const newArgv = {...argv};
+  for (const option in configObject) {
+    const wasValueSetOnCLI = argv[option] &&
+      argv[option] !== defaultValues[option];
+    if (wasValueSetOnCLI) {
+      log.debug(`Favoring CLI: ${option}=${argv[option]} over ` +
+        `configuration: ${option}=${configObject[option]}`);
+      continue;
+    }
+    if (!argv.hasOwnProperty(option) || defaultValues[option]) {
+      newArgv[option] = configObject[option];
+      log.debug(`Favoring configuration: ${option}=${configObject[option]} ` +
+        `over CLI: ${option}=${argv[option]}`);
+    }
+  }
+  return newArgv;
+}
+
+export function loadJSConfigFile(filePath: string): Object {
+  log.debug(`Loading JS config file: ${filePath}`);
+  try {
+    return requireUncached(filePath);
+  } catch (error) {
+    log.debug('Handling error:', error);
+    throw new UsageError(
+        `Cannot read config file: ${filePath}\nError: ${error.message}`);
+  }
```

Comment:
I think it would be nice to add a debug warning about empty configs here since we don't have a way to detect when a user forgets to type `module.exports = ...`. Something like:
````js
if (Object.keys(configObject).length === 0) {
  log.debug(
    `Config file ${filePath} did not define any options. ` +
    `Did you set module.exports = {...}?`);
}
````
I'm suggesting `log.debug()` not `log.warn()` because there could be legitimate cases where the config won't have any keys, such as when working with a template for a new project.

Because of code coverage, you will need to add a test that tries to load an empty config but that test won't need to make any assertions. It will just be a test to make sure no errors are thrown.